hierarc/Likelihood/LensLikelihood/ddt_hist_likelihood.py

```diff
@@ -25,25 +26,36 @@ def __init__(self, z_lens, z_source, ddt_samples, kde_kernel=None, ddt_weights=N
         :param kde_kernel: string of KDE kernel type
         :param bandwidth: bandwith of kernel
         :param nbins_hist: number of bins in the histogram
-        :param normalized: bool, if True, returns the normalized likelihood, if False, separates the constant prefactor
-         (in case of a Gaussian 1/(sigma sqrt(2 pi)) ) to compute the reduced chi2 statistics
+        :param normalized: bool, if True, returns the normalized likelihood, 
+        if False, separates the constant prefactor (in case of a Gaussian 
+        1/(sigma sqrt(2 pi))) to compute the reduced chi2 statistics
+        :binning_method: method used to calculate the bandwidth. "scott", 
+        "silverman", and a scalar constant (KDE factor) are supported. 
+        (See the scipy.stats.gaussian_kde documentation for details.)
         """
-
-        hist = np.histogram(ddt_samples, bins=nbins_hist, weights=ddt_weights)
-        vals = hist[0]
-        bins = [(h + hist[1][i + 1]) / 2.0 for i, h in enumerate(hist[1][:-1])]
-
-        # ignore potential zero weights, sklearn does not like them
-        kde_bins = [b for v, b in zip(vals, bins) if v > 0]
-        kde_weights = [v for v in vals if v > 0]
-        self._kde = gaussian_kde(dataset=kde_bins, weights=kde_weights[:])
+        if binning_method is None:
+            hist = np.histogram(ddt_samples, 
+                                bins=nbins_hist, 
+                                weights=ddt_weights)
+            vals = hist[0]
+            bins = [(h + hist[1][i+1])/2.0 for i, h in enumerate(hist[1][:-1])]
+            # ignore potential zero weights, sklearn does not like them
+            kde_bins = [b for v, b in zip(vals, bins) if v > 0]
+            kde_weights = [v for v in vals if v > 0]
+            self._kde = gaussian_kde(dataset=kde_bins, 
+                                     weights=kde_weights[:])
+        else:
+            self._kde = gaussian_kde(dataset=ddt_samples, 
```

Comment:
can you add a test for this option please?

Reply:
Tests added! I used unittest but pytest should still recognize it.